### DISCUSSION TITLE
Узнать плейлист по треку

### DISCUSSION THREAD

**Discussion by Bedmit** [Chimildic/goofy#33] [Идея] [answered]
Есть ли возможность вставлять в описательную часть создаваемого плейлиста, название плейлиста по которому он создавался? Наподобие как в getDescription вставляются треки, но только аргументом чтобы были не треки, а название плейлиста.

**Comment by Chimildic** [comment]
Я так понимаю речь о динамически меняющихся или заранее неизвестных плейлистах-источниках. Без правок библиотеки не выйдет. Это действительно нужно?

**Reply by Bedmit** [reply]
Ну в основном да, для динамических. Чтобы понимать на основе какого плейлиста они созданы. 
Но если это требует каких-то сложных телодвижений, то особой необходимости нет. Эта хотелка не так важна, как предыдущие.

**Comment by Chimildic** [comment, marked as answer]
Для версии 1.4.2+. Переменная `str` содержит имена всех уникальных плейлистов.
```js
let tracks = Source.getPlaylistTracks('', 'id');
let str = Array.from(new Set(tracks.map(t => t.origin.name))).join(', ');
```

**Comment by efremych** [comment]
Так как данная хотелка может помочь мне избавиться еще от ряда костылей, то вброшу еще, что помимо плейлистов можно было бы отслеживать такие источники как Топ треки артиста, Рекомендации Ласт ФМ, Рекомендации Спотифай и прочие функции, которые генерят треки.
Можно не отвечать, так как данный тикет решен :) просто вброс

**Reply by Chimildic** [reply]
Топ треки исполнителя содержат самого исполнителя. Зачем им источник?
Рекомендации могут появится только если использовать соответствующие функции. Без вставок понятно, что трек придет из них. Для составления описания плейлиста, каждый трек не должен содержать свой источник. И так можно дописать статичные слова про это.

**Reply by efremych** [reply]
Я вот про это, выделенные строки - это результат функций. Если для чего-то понадобились в описании итогового плейлиста плейлисты-исходники, то следующим шагом должны понадобиться и функции, которые дали треки для итогового плейлиста
![image](https://user-images.githubusercontent.com/76212904/107115837-5a003e00-6880-11eb-8775-b8b2625bd320.png)

**Reply by Chimildic** [reply]
Плейлисты в некоторых функциях заранее неизвестны. И в описание включить наверное интересно.
Но в чем ценность функций в описании? Если они есть в коде, собирающем плейлист. 
По каждому треку вообще экзотика какая-то) Что дает информация о том, что `Get Back` пришло из топа исполнителя? Как-то ни горячо, ни холодно от этого. Я чего-то не понимаю.

**Reply by efremych** [reply]
Насчет описаний не знаю, мне интересно собирать в файл.
Я, конечно, не скажу, что этим файлом регулярно пользуюсь - просто копится, обновляется.
Но, например, без него не обнаружил бы, что нарушилась пропорция, которую генерит `mixin`.
Также, если бы не собирал рекомендации Last FM (getNeighboursStation+getRecomStation) в отдельный плейлист, из которого пополнял итоговый плейлист. То мог бы определить откуда сыпется трэш в итоговый плейлист (оказалось, виной был getNeighboursStation).
Или кто-то. например, захочет треки нескольких радиостанций собирать в один плейлист - можно визуально определить какая радиостанция лишняя, потому что она генерит трэш, или нужно уменьшить количество треков от нее.
Да мало ли зачем еще))) анализ данных так сказать))))

Еще раз, ничего не прошу, просто подумал, что это логичное масштабирование добавленной конструкции :)
